Server/serverSlate.go

```diff
@@ -36,21 +59,49 @@ func main() {
 	fmt.Println("Got connection from:")
 	fmt.Println(conn.LocalAddr().String())
 
-	hostReader := bufio.NewReader(conn)
-	timeStamps := make([]string, KEY_PRESS_COUNT+1)
+	timestamps := make([]int64, KEY_PRESS_COUNT+1)
+
+	hostCommunicator := json.NewDecoder(conn)
 
-	message, _ := hostReader.ReadString('*') // using * as delim and hardcoded in app to append * at end of message
-	fmt.Println(message)
-	if message == "started capture"+"*" {
-		simulateKeyPress(KEY, KEY_PRESS_COUNT, timeStamps)
+	var hostAction action
+
+	for {
+		err := hostCommunicator.Decode(&hostAction)
+		fmt.Println(err)
+		if hostAction.Code == 1 {
+			simulateKeyPress(KEY, KEY_PRESS_COUNT, timestamps)
+		} else if hostAction.Code == 2 {
+			var ocrData hostData
+			hostCommunicator.Decode(&ocrData)
+			calculateLag(ocrData, timestamps)
+		}
 	}
-	fmt.Println("Key simulation ended")
+}
+
+func calculateLag(ocrData hostData, timestamps []int64) []int64 {
+	lagResults := make([]int64, KEY_PRESS_COUNT+1)
+	syncOffset := ocrData.HostSyncTimestamp - timestamps[0]
 
-	message, _ = hostReader.ReadString('*') // using * as delim and hardcoded in app to append * at end of message
-	fmt.Println(message)
-	if message == "send timestamps"+"*" {
-		sendTimeStamps(conn, timeStamps)
+	searchKey := ""
+	frameDuration := 1000 / ocrData.VideoFps
```

Comment:
maybe use a float, work in nanoseconds, or do the multiplicatoin by j before the division to avoid loss of precision here.